File: 05_functions/python/fizzbuzz.py
if __name__ == "__main__":
    # not possible as lambda
    def fizz_buzz(x):
        if x % 15 == 0:
            print("FizzBuzz")
        elif x % 3 == 0:
            print("Fizz")
        elif x % 5 == 0:
            print("Buzz")
        else:
            print(x)

    for i in range(1, 16):
        fizz_buzz(i)

    print("---")
    list(map(fizz_buzz, range(1, 16)))

File: 05_functions/rust/fizzbuzz.rs
fn main() {
    let fizz_buzz = |x| {
        if x % 15 == 0 {
            println!("FizzBuzz")
        } else if x % 3 == 0 {
            println!("Fizz")
        } else if x % 5 == 0 {
            println!("Buzz")
        } else {
            println!("{}", x)
        }
    };
    for i in 1..16 {
        fizz_buzz(i)
    }
    println!("---");
    (1..16).into_iter().for_each(fizz_buzz);
}
